platform/src/main/java/org/conscrypt/AndroidHpkeSpi.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.conscrypt;
+
+import android.crypto.hpke.HpkeSpi;
+
+import libcore.util.NonNull;
+import libcore.util.Nullable;
+
+import java.security.GeneralSecurityException;
+import java.security.InvalidKeyException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+
+/**
+ * Delegating wrapper for HpkeImpl that inherits the Android platform's SPI
+ * as well as Conscrypt's own.
+ */
+public class AndroidHpkeSpi implements HpkeSpi, org.conscrypt.HpkeSpi {
```

Comment:
Can we use just a single HpkeSpi?  I mean this class just seems to have an indirection to org.conscrypt.HpkeSpi, but I don't know for sure why it's needed.

Reply:
I assume the second "implements" should be android.crypto.hpke.HpkeSpi.